src/vma/dev/ring_tap.cpp

```diff
@@ -1023,9 +1025,13 @@ int ring_tap::process_element_rx(void* pv_fd_ready_array)
 			if (ret > 0) {
 				/* Data was read and processed successfully */
 				buff->sz_data = ret;
-				rx_process_buffer(buff, pv_fd_ready_array);
-				m_p_ring_stat->tap.n_rx_buffers--;
-			} else {
+				buff->rx.is_sw_csum_need = 1;
+				if (rx_process_buffer(buff, pv_fd_ready_array)) {
+					m_p_ring_stat->tap.n_rx_buffers--;
+					used_buff = true;
+				}
+			}
+			if (!used_buff){
```

Comment:
please consider returning number of processed buffers in `return ret;` to be consistent with ring simple.
```
if (ret > 0) {
...
ret = rx_process_buffer(buff, pv_fd_ready_array);
}

if (ret > 0) {
    m_p_ring_stat->tap.n_rx_buffers--;
} else {
    ret = 0;
    m_rx_pool.push_front(buff);
}
```